### ISSUE TITLE
CUDA no kernel image available, on Linux with Docker

### ISSUE THREAD

**Issue by dougbtv** [oobabooga/text-generation-webui#2002]
### Describe the bug

I've built the docker image, and the webui runs -- however, when I ask to generate a prompt from the webui, I'm getting:

```
RuntimeError: CUDA error: no kernel image is available for execution on the device
```

I've tried modifying the TORCH_CUDA_ARCH_LIST in the .env file for docker compose, to use 8.0 or 8.6+PTX, all the versions, and also commenting out the line, all with the same result.

I've also checked which version of torch is installed...

```
(venv) root@316f2ad5c074:/app# pip list | grep -i torch
torch                    2.0.1
torchaudio               2.0.2
torchvision              0.15.2
```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

* docker-compose build / run with a 3090 on Fedora 38

### Logs

```shell
(venv) root@355268d13866:/app# python server.py --model TheBloke_stable-vicuna-13B-GPTQ --model_type=Llama --wbits 4 --groupsize 128 --listen --auto-devices
INFO:Gradio HTTP request redirected to localhost :)
bin /app/venv/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda118.so
INFO:Loading TheBloke_stable-vicuna-13B-GPTQ...
INFO:Found the following quantized model: models/TheBloke_stable-vicuna-13B-GPTQ/stable-vicuna-13B-GPTQ-4bit.compat.no-act-order.safetensors
INFO:Loaded the model in 4.67 seconds.

Running on local URL:  http://0.0.0.0:7860

To create a public link, set `share=True` in `launch()`.
Traceback (most recent call last):
  File "/app/modules/callbacks.py", line 73, in gentask
    ret = self.mfunc(callback=_callback, **self.kwargs)
  File "/app/modules/text_generation.py", line 259, in generate_with_callback
    shared.model.generate(**kwargs)
  File "/app/venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/app/venv/lib/python3.10/site-packages/transformers/generation/utils.py", line 1485, in generate
    return self.sample(
  File "/app/venv/lib/python3.10/site-packages/transformers/generation/utils.py", line 2524, in sample
    outputs = self(
  File "/app/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/app/venv/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 687, in forward
    outputs = self.model(
  File "/app/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/app/venv/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 577, in forward
    layer_outputs = decoder_layer(
  File "/app/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/app/venv/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 292, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "/app/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/app/venv/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 196, in forward
    query_states = self.q_proj(hidden_states).view(bsz, q_len, self.num_heads, self.head_dim).transpose(1, 2)
  File "/app/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/app/repositories/GPTQ-for-LLaMa/quant.py", line 431, in forward
    y = y.to(output_dtype)
RuntimeError: CUDA error: no kernel image is available for execution on the device
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

### System Info

```shell
Fedora 38
Nvidia 3090
```

**Comment by dougbtv** [comment]
For what it's worth I do see my GPU when I run:

```
root@626d2c759410:/app# . ./venv/bin/activate
(venv) root@626d2c759410:/app# python3
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.cuda.is_available()
True
>>> torch.cuda.device_count()
1
>>> torch.cuda.get_device_name(0)
'NVIDIA GeForce RTX 3090'
>>> torch.tensor([1.0, 2.0]).cuda()
tensor([1., 2.], device='cuda:0')
```

I also tried to specifically install the `+cu118` editions of torch by changing the lines for those install for GPTQ and the webui in the dockerfile... same result, though.

```
```
-    pip3 install torch torchvision torchaudio && \
+    pip3 install torch==2.0.1+cu118 torchvision==0.15.2+cu118 torchaudio==2.0.2 -f https://download.pytorch.org/whl/cu118/torch_stable.html && \
```

```

**Comment by dougbtv** [comment]
I was able to get it to work. I think the main cause might have not been setting the correct 

I kept my changes for torch (so far) but I also found that it worked better when I found the proper settings

I double checked the version with [the cuda version / gpu model chart on wikipedia](https://en.wikipedia.org/wiki/CUDA#GPUs_supported) and found that I should use solely 8.6 for the 3090. 

I had tried just 8.0 and 8.6 which may have worked -- but because of the weird symlinking of the dockerfile/compose, I have cross references between build and run somehow, and sometimes it's using "docker/" folder and sometimes the "./" root folder. So I may have had failed tests because it wasn't properly being set.

I set in the .env to use at build time as such:

```
TORCH_CUDA_ARCH_LIST=8.6+PTX
```
